doc/ref/chap-type-method.md

```diff
@@ -27,6 +27,25 @@ This chapter in the [Oils Reference](index.html) describes YSH types and methods
 
 ### replace()
 
+Replace substrings or eggex matches within a string.
+
+    var mystr = "code is 1234"
+    = mystr=>replace("code", "Code")  # => "Code is 1234"
+    = mystr=>replace(/ d+ /, "<redacted>")  # => "code is <redacted>"
+
+Replacement expressions can use captures from the given eggex.
+
+    var mystr = "year is 1988"
+    = mystr=>replace(/ <capture digit{4} as year : int> /, ^"$[year + 1]")
```

Comment:
This should also document positional captures like $1 $2

and $0 when we implement it